lib/Parse/Lexer.cpp

```diff
@@ -1380,7 +1405,11 @@ static const char *skipToEndOfInterpolatedExpression(const char *CurPtr,
           // Entering a recursive interpolated expression
           OpenDelimiters.push_back('(');
           continue;
-        case '\n': case '\r': case 0:
+        case ' ': case '\t': case '\n': case '\r':
+          if (MultilineString && maybeConsumeNewlineEscape(CurPtr, -1))
+            continue;
+          LLVM_FALLTHROUGH;
```

Comment:
I think this would be sufficient:
```cpp
        case '\n': case '\r':
          if (AllowNewline.back())
            continue;
          LLVM_FALLTHROUGH;
        case 0:
          // Don't jump over newline/EOF due to preceding backslash!
          return CurPtr-1;
```